spec/schemas/_common.aggregations.yaml

```diff
@@ -114,10 +114,10 @@ components:
       type: object
       additionalProperties:
         oneOf:
-          - type: string
           - type: number
-          - nullable: true
-            type: string
+          - type:
+              - 'null'
+              - string
```

Comment:
What does this actually mean?

```
oneOf:
  - type: number
  - type:
     - 'null'
     - string
```

Is it not the same as 

```
type:
  - oneOf: 
    - number
    - string
    - 'null'
```

or even

```
type: [number, string, 'null']
```

?

Reply:
If my research is correct then both following definitions are valid.
```
 oneOf:
    - type: 'null'
    - type: number
    - type: string
```
```
type:
    - 'null'
    - number
    - string
``` 

According to https://blog.stoplight.io/difference-between-open-v2-v3-v31
> With OpenAPI 3.1, the specification now supports type as an array

Reply:
Should I change all of those to the second definition with type as an array?

Reply:
I don't think that's necessary, but collapsing the nested structure probably is.

Reply:
```yaml
type:
    - 'null'
    - number
    - string
```
Is the best option IMO.

Reply:
I think it would be clearer if we created a type for `NullableString` because `null` can be ambiguous if you're not paying attention. I'm ok with what @nhtruong proposes though, @Jakob3xD your call.

Reply:
`null` without quotes is not valid and will fail the validation test.
https://github.com/Jakob3xD/opensearch-api-specification/actions/runs/10003423255/job/27650310114

I'll do the suggested type array.

Reply:
When using the type array we need to set allowUnionTypes for Ajv but I am not sure where I can configure it.